compiler/rustc_mir_build/src/check_unsafety.rs

```diff
@@ -449,6 +449,20 @@ impl<'a, 'tcx> Visitor<'a, 'tcx> for UnsafetyVisitor<'a, 'tcx> {
                     }
                 }
             }
+            ExprKind::AddressOf { arg, .. } => {
+                if let ExprKind::Scope { value: arg, .. } = self.thir[arg].kind
+                // THIR desugars UNSAFE_STATIC into *&raw mut UNSAFE_STATIC, so take two steps
+                    && let ExprKind::Deref { arg } = self.thir[arg].kind
+                    && let ExprKind::StaticRef { .. } | ExprKind::ThreadLocalRef { .. } = self.thir[arg].kind
+                {
+                    // a raw ref to a place expr, even an "unsafe static", is okay!
+                    return;
+                    // note: const_mut_refs enables this code, and it currently remains unsafe:
```

Comment:
Specifically the 2nd `unsafe` remains, but the first goes away, right?

Reply:
Yeah, drafted a test to go into detail on that.